src/main/java/io/github/jelastic/core/elastic/ElasticClient.java

```diff
@@ -76,6 +87,20 @@ public ElasticClient(JElasticConfiguration configuration) throws IOException {
         RestClientBuilder restClientBuilder = RestClient.builder(jElasticConfiguration.getServers().stream().map(hostAndPort -> new HttpHost(hostAndPort.getHost(), hostAndPort.getPort())).toArray(HttpHost[]::new));
 
         if(null != jElasticConfiguration.getAuthConfiguration()){
+
+            if(jElasticConfiguration.getAuthConfiguration().isTlsEnabled()){
+                Path trustStorePath = Paths.get(jElasticConfiguration.getAuthConfiguration().getTrustStorePath());
+                KeyStore truststore = KeyStore.getInstance("pkcs12");
+                try (InputStream is = Files.newInputStream(trustStorePath)) {
+                    truststore.load(is, jElasticConfiguration.getAuthConfiguration().getKeyStorePass().toCharArray());
+                }
+                SSLContextBuilder sslBuilder = SSLContexts.custom()
+                        .loadTrustMaterial(truststore, null);
+                final SSLContext sslContext = sslBuilder.build();
+                restClientBuilder.setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder.setSSLContext(sslContext));
```

Comment:
You can combine both the lines setSSLContext and restClientBuilder.setCredentials into one lamdda, and declare the variables atop with a null check. Possibly extract out these two into different methods getSSLContext and getCredentials; and if they are not null, then apply to the clientBuilder.

Reply:
this has been taken care of !